### PR TITLE
WIP fix(ci): experiments on how to make CI run faster

### PR DESCRIPTION
**Pull request by fredbi**
1. Make linting startup with minimal overhead

TODO:
2. Parallel unit test [hit bug here]
3. build images earlier in proces?

Signed-off-by: Frederic BIDON <frederic@oneconcern.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ jobs:
             export GOOGLE_APPLICATION_CREDENTIALS=$HOME/extra/appcredentials.json
             hack/go-generate.sh
             go mod download
-            gotestsum --junitfile /tmp/test-results/go-test-report.xml ./... -- -race ./...
+            gotestsum --junitfile /tmp/test-results/go-test-report.xml ./... -- ./...
```
Comment (ransomw1c) [inline .circleci/config.yml]: on the surface of it, i suppose the race detector is a desirable feature of the test runner, despite additional time?

Comment (fredbi) [reply .circleci/config.yml]: yes I just wanted to gauge the extra time needed by race and possibly run a the test with race in a separate workflow, not required by image builds.

Comment (fredbi) [reply .circleci/config.yml]: benefit from removing race is negligible. Putting it again.

**Comment by kerneltime** [comment]
Please park this branch for now and close this PR.

**Comment by fredbi** [comment]
Parked. This is mostly superseded by #428

**Pull request closed without merging**